lib/optimist.rb

```diff
@@ -316,10 +331,14 @@ def parse(cmdline = ARGV)
         else                       raise CommandlineError, "invalid argument syntax: '#{arg}'"
       end
 
-      sym = nil if arg =~ /--no-/ # explicitly invalidate --no-no- arguments
+      if arg =~ /--no-/ # explicitly invalidate --no-no- arguments
+        sym = nil
+      ## Support inexact matching of long-arguments like perl's Getopt::Long
+      elsif !sym && !@settings[:exact_match] && arg.match(/^--(\S*)$/)
```

Comment:
```suggestion
      elsif !sym && !@settings[:exact_match] && arg.match(/^--(\S+)$/)
```

Otherwise this would match on `--` by itself.